aws-service-discovery/src/main/java/io/micronaut/discovery/aws/servicediscovery/AwsServiceDiscoveryClientConfiguration.java

```diff
@@ -0,0 +1,122 @@
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.discovery.DiscoveryConfiguration;
+import io.micronaut.discovery.client.DiscoveryClientConfiguration;
+import io.micronaut.discovery.registration.RegistrationConfiguration;
+
+/**
+ * Configuration for AWS Route 53 discovery.
+ *
+ * @author Rvanderwerf
+ * @since 1.0
+ * See https://docs.aws.amazon.com/Route53/latest/APIReference/overview-service-discovery.html for details info
+ */
+@Internal
+@Requires(env = Environment.AMAZON_EC2)
+@Requires(property = AwsServiceDiscoveryConfiguration.ENABLED, value = StringUtils.TRUE, defaultValue = StringUtils.FALSE)
+@ConfigurationProperties(AwsServiceDiscoveryClientConfiguration.PREFIX)
+@BootstrapContextCompatible
+public class AwsServiceDiscoveryClientConfiguration extends DiscoveryClientConfiguration {
+
+    public static final String PREFIX = AwsServiceDiscoveryConfiguration.PREFIX + ".client";
+
+    private String awsServiceId; //ID of the service - required to find it
+    private String namespaceId; // used to filter a list of available services attached to a namespace
+
+    /**
+     * This is the name space ID for the domain/subdomain in route 53 service registry.
+     * You must have this defined before doing any operations on using service registry.
+     *
+     * @return namespace id
+     */
+    public String getNamespaceId() {
+        return namespaceId;
+    }
+
+    /**
+     * allows you to override the namespace id for testing.
+     *
+     * @param namespaceId name space id used for easy of testing
+     */
+    public void setNamespaceId(String namespaceId) {
+        this.namespaceId = namespaceId;
+    }
+
+    /**
+     * service ID where you are going to be adding service instances to.
+     *
+     * @return the service ID we are working with
+     */
+    @Override
+    protected String getServiceID() {
+        return awsServiceId;
+    }
+
+    /**
+     * Gets the discovery configuration.
+     *
+     * @return configuration
+     */
+    @NonNull
+    @Override
+    public DiscoveryConfiguration getDiscovery() {
+        return new AwsServiceDiscoveryConfiguration();
+    }
+
+    /**
+     * Get the registration configuration needed to register to service registry.
+     *
+     * @return configuration
+     */
+    @Nullable
+    @Override
+    public RegistrationConfiguration getRegistration() {
+        return null;
+    }
+
+    /**
+     * service id registered to aws.
+     *
+     * @return service id
+     */
+    public String getAwsServiceId() {
+        return awsServiceId;
+    }
+
+    /**
+     * set service id for easier testing.
+     *
+     * @param awsServiceId service id
+     */
+    public void setAwsServiceId(String awsServiceId) {
+        this.awsServiceId = awsServiceId;
+    }
+
+    @Override
+    public ConnectionPoolConfiguration getConnectionPoolConfiguration() {
+        return null;
```

Comment:
Is this supposed to return `null`?

Reply:
It's a copy of `Route53ClientDiscoveryConfiguration` and it looks like pooling is not supported for the client.